### DISCUSSION TITLE
UnoCSS HMR with HTML-only mode

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/vike#927] [Discussion]
UnoCSS's HMR doesn't work when using vite-plugin-ssr with [HTML-only](https://vite-plugin-ssr.com/render-modes#html-only) mode.

Note that UnoCSS (including HMR) does work when using vite-plugin-ssr with SPA and SSR mode.

The problem is fixable (see https://github.com/unocss/unocss/issues/1351#issuecomment-1207175309). It's just low priority because HTML-only is more of an experiment than being used in production.

**Comment by AaronBeaudoin** [comment]
I don't think you should need to write any specific integration to support UnoCSS. See my new discussion #412.

**Comment by brillout** [comment]
UnoCSS is rightfully loaded in the browser, that's not the problem. Your proposal wouldn't be helpful for this specific ticket.

The problem and the fix is about how UnoCSS is injected into the HTML.

I expect the fix to be fairly easy, actually. But because HTML Templating is low priority, I'm de-prioritizing this.

I'm open to re-evaluating the prioritization though, e.g. if it's a blocker for a project of yours.
